src/include/mpir_request.h

```diff
@@ -359,9 +359,6 @@ static inline MPIR_Request *MPIR_Request_create(MPIR_Request_kind_t kind, int po
 #define MPIR_Request_add_ref(req_p_) \
     do { MPIR_Object_add_ref(req_p_); } while (0)
 
-#define MPIR_Request_release_ref(req_p_, inuse_) \
```

Comment:
We have lots of `release_ref` wrappers in various `mpir_*.h` files. I guess in this case this one is only used in a single instance in this file, and `MPIR_Request_free` is  really what we intend users to call in the code. I suppose I've talked myself into approving it...

Reply:
The wrapper macros are helpful for grepping when we have to find places where references to objects from a particular type of class are released. Just faced this use case :)

Reply:
Which text were you trying to grep?

Reply:
`MPIR_Request_release_ref`

Reply:
No where is using it anyway, so the only place you would find is here :)

Reply:
Right, but I cannot find that using grep without the wrapper. Plus it is inconsistent with the rest of the codebase.

Reply:
You mean "rest of the normal handle objects"? Now we added request pool, it is different from all the rest handle objects :)

Reply:
We can add a comment that include that text for consistency of grepping, would that help?

Reply:
`MPIR_Request_add_ref` still exists but now there is no corresponding `MPIR_Request_release_ref`. That is inconsistent. Also, other classes (`MPII_Keyval`, `MPIR_Comm`, `MPIR_Datatype`, etc.) have similar wrappers. We are inconsistent with that.

The cleaner change would be to revert 634f5b2. It is not changing any functionality anyways. Adding a comment is just working around the changes of the commit.

Reply:
If you insist :) Go ahead open a PR for that, I'll approve it.